src/components/stepper/step/step.ts

```diff
@@ -0,0 +1,157 @@
+import { AgnosticResizeObserver } from '../../core/observers';
+import type { SbbStepLabelElement } from '../step-label';
+import type { SbbStepperElement } from '../stepper';
+
+import style from './step.scss?lit&inline';
+
+let nextId = 0;
+
+export type SbbStepValidateEventDetails = {
+  currentIndex?: number;
+  currentStep?: SbbStepElement;
+  nextIndex?: number;
+  nextStep?: SbbStepElement;
+};
+
+/**
+ * Combined with a `sbb-stepper`, it displays a step's content.
+ *
+ * @slot - Use the unnamed slot to provide a content.
+ * @event {CustomEvent<SbbStepValidateEventDetails>} validate - Emits whenever step switch is triggered. Can be canceled.
+ */
+@customElement('sbb-step')
+@hostAttributes({
+  slot: 'step',
+  role: 'tabpanel',
+})
+export class SbbStepElement extends LitElement {
+  public static override styles: CSSResultGroup = style;
+  public static readonly events = {
+    validate: 'validate',
+  } as const;
+
+  /**
+   * Emits whenever step switch is triggered.
+   */
+  private _validate: EventEmitter<SbbStepValidateEventDetails> = new EventEmitter(
+    this,
+    SbbStepElement.events.validate,
+  );
+
+  private _abort = new ConnectedAbortController(this);
+  private _stepper: SbbStepperElement | null = null;
+  private _label: SbbStepLabelElement | null = null;
+  private _stepResizeObserver = new AgnosticResizeObserver((entries) =>
+    this._onStepElementResize(entries),
+  );
+
+  /**
+   * The label of the step.
+   */
+  public get label(): SbbStepLabelElement | null {
+    return this._label;
+  }
+
+  /**
+   * Selects and configures the step.
+   * @internal
+   */
+  public select(): void {
+    if (!this.label) {
+      return;
+    }
+    this.toggleAttribute('data-selected', true);
+    this.label.select();
+  }
+
+  /**
+   * Deselects and configures the step.
+   * @internal
+   */
+  public deselect(): void {
+    if (!this.label) {
+      return;
+    }
+    this.toggleAttribute('data-selected', false);
+    this.label.deselect();
+  }
+
+  public validate(eventData: SbbStepValidateEventDetails): boolean {
+    return !!this._validate.emit(eventData);
+  }
+
+  private _handleClick(event: Event): void {
+    const composedPathElements = event
+      .composedPath()
+      .filter((el) => el instanceof window.HTMLElement);
+    if (composedPathElements.some((el) => this._isGoNextElement(el as HTMLElement))) {
+      this._stepper?.next();
+    } else if (composedPathElements.some((el) => this._isGoPreviousElement(el as HTMLElement))) {
+      this._stepper?.previous();
+    }
+  }
+
+  private _isGoNextElement(element: HTMLElement): boolean {
+    return element.hasAttribute('sbb-stepper-next') && !isValidAttribute(element, 'disabled');
+  }
+
+  private _isGoPreviousElement(element: HTMLElement): boolean {
+    return element.hasAttribute('sbb-stepper-previous') && !isValidAttribute(element, 'disabled');
+  }
+
+  private _onStepElementResize(entries: ResizeObserverEntry[]): void {
```

Comment:
does a slotchange (changed content) trigger a resize too? otherwise we would need to respect the slotchange event as well I think

Reply:
yes, if I change the content the step will be resized accordingly.